File: GaragEZ/src/types/part.ts
export interface Part {
  id?: number;  // 선택적 필드 (추가 시 자동 생성)
  name: string;  // 부품 이름
  category: string;  // 부품 카테고리
  price: number;  // 부품 가격
  stock: number;  // 부품 재고
  imageUrl?: string;  // 부품 이미지 URL (선택적)
}

File: BackEnd/src/main/java/com/company/entity/part/Part.java
package com.company.entity.part;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import java.time.LocalDateTime;

@Entity
@Table(name = "parts")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Part {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String category;
    private double price;
    private int stock;

    @CreationTimestamp
    @Column(name = "created_at")
    private LocalDateTime createdAt;

    private String imageUrl;  // 부품 이미지 URL (선택적 필드 추가)
}
